cpp/src/CMakeLists.txt

```diff
@@ -246,6 +240,20 @@ add_dependencies(spark_columnar_jni jni_proto)
 find_arrow_lib(${ARROW_LIB_NAME})
 find_arrow_lib(${GANDIVA_LIB_NAME})
 
+if(ENABLE_HBM)
```

Comment:
Do we need to install the memkind library? or build from source code?
If we need to install memkind library, let's add it to dependency list in doc.